### PR TITLE
RHDEVDOCS-4110: GitOps Operator to configure secure communication wit…

### PR DESCRIPTION
**Pull request by eromanova97**
• **Aligned team**: Dev Tools
• **OCP version for cherry-picking**: enterprise-4.10 and later
• **JIRA issue**: [RHDEVDOCS-4110](https://issues.redhat.com/browse/RHDEVDOCS-4110)
• **Preview page**: [Configuring secure communication with Redis](https://62228--docspreview.netlify.app/openshift-enterprise/latest/cicd/gitops/configuring-secure-communication-with-redis.html)
• **SME Review**: Completed by @iam-veeramalla 
• **QE review**: Completed by @varshab1210 
• **Peer-review**: Completed by @Srivaralakshmi

### CODE REVIEW

**Comment by ekristova** [comment]
Hello @iam-veeramalla @varshab1210 PTAL and provide feedback. Thank you!

**Review by iam-veeramalla — APPROVED** [review]
/lgtm

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+The setting with HA enabled requires a cluster with at least three worker nodes.
+====
+
+.. Click *Create*.
+
+.. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.
++
+[source,terminal]
+----
+$ oc get pods
```
Comment (varshab1210) [inline modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:48]:
```suggestion
$ oc get pods -n openshift-gitops
```

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+[source,terminal]
+----
+secret/argocd-operator-redis-tls annotated
+----
+
+. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.
++
+[source,terminal]
+----
+$ oc get pods
```
Comment (varshab1210) [inline modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:198]:
```suggestion
$ oc get pods -n openshift-gitops
```

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+The setting with HA enabled requires a cluster with at least three worker nodes.
+====
+
+.. Click *Create*.
+
+.. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.
++
+[source,terminal]
+----
+$ oc get pods
```
Comment (varshab1210) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:66]:
```suggestion
$ oc get pods -n openshift-gitops
```

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+argocd-redis-ha-server-2                   2/2     Running   0          53s
+argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
+argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
+----
+
+. Verify that a secret called `argocd-operator-redis-tls` was created:
++
+[source,terminal]
+----
+$ oc get secrets argocd-operator-redis-tls
```
Comment (varshab1210) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:96]:
```suggestion
$ oc get secrets argocd-operator-redis-tls -n openshift-gitops
```

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+.. In the YAML field, replace the YAML code:
++
+.Example ArgoCD CR with autotls disabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: argocd <1>
+  namespace: openshift-gitops <2>
```
Comment (varshab1210) [inline modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:27]: @iam-veeramalla Is it okay to mention the namespace as **openshift-gitops** as the name of argocd nstance given here is **argocd** whereas we already have an instance **openshift-gitops** in that namespace?

Comment (iam-veeramalla) [reply modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:27]: @varshab1210 I think that is ok, as in the description it is clearly mentioned what name and namespace represents.

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc (hunk 1 above)
Comment (ekristova) [reply modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc]: Hello @varshab1210 as I did not want to create commands for specific namespaces, I incorporated your suggestion by adding
`-n <namespace>`
I hope you are okay with that change. I added it to the places you suggested. Please let me know if the PR looks good to you now. Thank you!

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+====
++
+[NOTE]
+====
+The setting with HA enabled requires a cluster with at least three worker nodes.
+====
+
+.. Click *Create*.
+
+.. Verify that the Argo CD pods are ready and running. For the Argo CD instance with HA enabled, this might take a few minutes.
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:61]: @ekristova Please follow the suggestion I gave above for a similar instance. I think it is for substep f of step 1 in the Configuring TLS for Redis with autotls disabled section for this portion of content.

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+argocd-application-controller-0            1/1     Running   0          10m
+argocd-redis-ha-haproxy-669757fdb7-5xg8h   1/1     Running   0          10m
+argocd-redis-ha-server-0                   2/2     Running   0          9m9s
+argocd-redis-ha-server-1                   2/2     Running   0          98s
+argocd-redis-ha-server-2                   2/2     Running   0          53s
+argocd-repo-server-576499d46d-8hgbh        1/1     Running   0          10m
+argocd-server-9486f88b7-dk2ks              1/1     Running   0          10m
+----
+
+. Verify that a secret called `argocd-operator-redis-tls` was created:
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:92]:
```suggestion
. Verify that a `argocd-operator-redis-tls` secret is created:
```

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@ekristova Splendid job! Excellent work! Thank you very much.

PTAL at the suggestions.

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+  -reqexts SAN -extensions SAN \
+  -config <(printf "\n[SAN]\nsubjectAltName=DNS:argocd-redis.<namespace>.svc.cluster.local\n[req]\ndistinguished_name=req") \ <1>
+  -keyout /tmp/redis.key \
+  -out /tmp/redis.crt \
+  -newkey rsa:4096 \
+  -nodes \
+  -sha256 \
+  -days 10
+----
+<1> The namespace where the Argo CD instance is running.
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:98]: @ekristova Revert to the original call-out sentences for commands. The original sentence is the method suggested in OCP doc guidelines too. This one is not at all clear to the user when performing the procedure. That is why I did not comment on callouts under commands. This is Global Change.

Only for callout sin examples do you follow this pattern. NOT for commands.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@ekristova Please address this comment: https://github.com/openshift/openshift-docs/pull/62228#discussion_r1273395731

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc (hunk 8 above)
Comment (ekristova) [reply modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:98]: I see, thank you for letting me know!

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc [hunk 9]
```diff
@@ -0,0 +1,235 @@
+// Module is included in the following assemblies:
+//
+// * /cicd/gitops/configuring-secure-communication-with-redis.adoc
+
+[id="gitops-configuring-tls-for-redis-with-autotls-disabled_{context}"]
+= Configuring TLS for Redis with autotls disabled
+
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:7]: @ekristova You need to add an introductory sentence or two for this procedure module file. See here for details on the different parts of a procedure module: https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+// * /cicd/gitops/configuring-secure-communication-with-redis.adoc
+
+[id="gitops-configuring-tls-for-redis-with-autotls-enabled_{context}"]
+= Configuring TLS for Redis with autotls enabled
+
+[NOTE]
+====
+By default, the `autotls` setting is disabled.
+====
+
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:12]: @ekristova You need to add an introductory sentence or two for this procedure module file. See here for details on the different parts of a procedure module: https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+<1> Specify a namespace where the Argo CD instance is running, for example `openshift-gitops`.
++
+.Example output 
+[source,terminal]
+----
+NAME                        TYPE                DATA   AGE
+argocd-operator-redis-tls   kubernetes.io/tls   2      30s
+----
++
+The secret should be the `kubernetes.io/tls` type and should have a size of `2`.
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:114]:
```suggestion
The secret must be the `kubernetes.io/tls` type and a size of `2`.
```
@ekristova Per the ISG: "Do not use should to refer to an action that must be performed." https://www.ibm.com/docs/en/ibm-style?topic=word-usage#s

@reginapizza and @varshab1210 PTAL and confirm that my suggestion does not change the technical accuracy. Thanks!

File: modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc (hunk 9 above)
Comment (ekristova) [reply modules/gitops-configuring-tls-for-redis-with-autotls-disabled.adoc:7]: Hello @Srivaralakshmi thank you for your observations!

As per the guideline for the procedure module, I realized I included part of that information in the introduction of the assembly, more specifically, in that two-point list providing the two methods. Therefore, I made the two points more brief and added the information from there to the introduction of each procedural module, plus I added those special considerations mentioned in the guideline. So PTAL and let me know what you think. 

Also, regarding the "should" issue in the comment below, I changed the wording that in my opinion has a clear meaning and is according to the guidelines. So let me know what you think about that one.

Thank you!

File: cicd/gitops/configuring-secure-communication-with-redis.adoc [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+:context: configuring-secure-communication-with-redis
+
+toc::[]
+
+{gitops-title} is able to use the Transport Layer Security (TLS) encryption to secure the communication between the Argo CD components and Redis cache so that possibly sensitive data are protected in transit.
+
+If you configure TLS for Redis with the `autotls` setting, no further steps are necessary. If you create the `argocd-operator-redis-tls` secret manually with the key and certificate pair, it requires an annotation to indicate it is owned by the appropriate Argo CD instance. Both the configurations are possible with or without the high availability (HA) enabled.
+
+.Prerequisites
+* {gitops-title} Operator is installed on your cluster.
```
[the remark this answers is not in the document]
Comment (varshab1210) [reply cicd/gitops/configuring-secure-communication-with-redis.adoc:21]: This should be okay

File: modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+<1> Specify a namespace where the Argo CD instance is running, for example `openshift-gitops`.
++
+.Example output 
+[source,terminal]
+----
+NAME                        TYPE                DATA   AGE
+argocd-operator-redis-tls   kubernetes.io/tls   2      30s
+----
++
+Ensure that the secret is the `kubernetes.io/tls` type and a size of `2`.
```
Comment (Srivaralakshmi) [inline modules/gitops-configuring-tls-for-redis-with-autotls-enabled.adoc:116]:
```suggestion
The secret must be of the `kubernetes.io/tls` type and a size of `2`.
```

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@ekristova PTAL at my suggestions. Thanks for the revision.

**Review by Srivaralakshmi — APPROVED** [review]
@ekristova Thanks for all the quality work! Approved.

**Review by jldohmann — APPROVED** [review]
LGTM

**Pull request merged**